FreshRSS/FreshRSS
# 2297 cli parameters consistency

Closes #2297

# Changes proposed in this pull request:

We want to standardise flags used with CLI commands around the use of '-' rather than '\_' while maintaining backward compatibility. To that end I've deprecated all flags using '\_' in favour of an equivalent using '-' and introduced a check for the use of deprecated flags as well as logic to replace any used deprecated flags with their replacements. I've also scooped up some existing logic for CLI flag retrieval and validation and wrapped all this up into a parseCliParams() function. This way we can start to isolate some logic that otherwise is scattered throughout individual CLI commands.

If we're happy with the changes proposed here, then I could look at raising some related tickets such as extending use of parseCliParams() to other relevant CLI commands and looking at other opportunities for standardisation around validating CLI inputs as there is currently quite a bit of variation in how individual CLI commands handle similar work e.g. validating options associated with flags or updating config.

# How to test the feature manually:

1. Use either cli/do-install, cli/reconfigure, cli/create-user or cli/update-user with updated flags (e.g. --default-user admin --allow-anonymous) to see successful behaviour.
![image](https://github.com/FreshRSS/FreshRSS/assets/59581846/20df6a64-206b-4764-85d8-8932afe8afc0)
![image](https://github.com/FreshRSS/FreshRSS/assets/59581846/aa1b5a85-dcff-42bf-8dcb-539ffbaebcf7)

2. Use deprecated flags (e.g. --default_user admin --allow_anonymous) to see successful behaviour with a deprecation warning.
![image](https://github.com/FreshRSS/FreshRSS/assets/59581846/7c54baa7-78f9-4557-892a-d516c4ae3cf0)
![image](https://github.com/FreshRSS/FreshRSS/assets/59581846/6aa22b2e-6caf-499c-8d8d-af53b940b442)


3. Use unknown flags (e.g. --defaultuser admin --allow_anon)to trigger expected warning and failure.
![image](https://github.com/FreshRSS/FreshRSS/assets/59581846/d4c261ab-ee0a-4254-95e0-0856808c9c0e)
![image](https://github.com/FreshRSS/FreshRSS/assets/59581846/7c4df5e1-204a-4df5-9eb8-33f5aeeb13f8)

# Pull request checklist:
